developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -129,132 +101,308 @@ Once it has been instantiated, you will notice that the client API is different
 
-  - talk a bit more about this and package specifics 
-  - Introduction of gRPC?
+For Web use, we recommend using the `v2` client or use the `v3` [web client](#).
 
 ### Improved Functionality
 
-(with examples)
+#### Improved Bulk Insertion
+We've added an `insertMany()` method as a replacement for `objectBatcher()` to help make batch insertions easier. 
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+const questions = client.collections.get("CollectionName")
+const dataObject = [...]; // your data
+
+await questions.data.insertMany(dataBatch);
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+let className = 'CollectionName';  // Replace with your collection name
+let dataObjs = [...];
+
+let batcher5 = client.batch.objectsBatcher();
+for (const dataObj of dataObjs)
+  batcher5 = batcher5.withObject({
+    class: className,
+    properties: dataObj,
+  });
+
+// Flush
+await batcher5.do();
+```
+
+</TabItem>
+</Tabs>
+
+For more information on batching, I recommend looking at the [batching section](typescript-v3.md#) of our client page.
+
+
+#### Client Close Method
+
+Using gRPC for CRUD operations, this client tends to keep its connection with Weaviate open longer that you're used to. We recommend closing this connection to save on resources with our `client.close()` method.
+
+#### Better Filter System
+
+We've made it easier to use filters with conditions by adding a `Filter` helper class. The overall experience of filters has been streamlined for cleaner more concise development. 
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+import weaviate, { Filters } from 'weaviate-client';
+const myCollection = client.collections.get('JeopardyQuestion');
+     
+const result = await myCollection.query.fetchObjects({
+  returnProperties: ['question', 'answer','round', 'points'],
+  filters: Filters.and(
+     myCollection.filter.byProperty('round').equal('Double Jeopardy!'),
+     myCollection.filter.byProperty('points').lessThan(600)
+    ),
+  limit: 3,
+ })
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+result = await client.graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withWhere({
+    operator: 'And',
+    operands: [
+      {
+        path: ['round'],
+        operator: 'Equal',
+        valueText: 'Double Jeopardy!',
+      },
+      {
+        path: ['points'],
+        operator: 'LessThan',
+        valueInt: 600,
+      },
+    ],
+  })
+  .withLimit(3)
+  .withFields('question answer round points')
+  .do();
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
+</Tabs>
+
+#### New Generate Namespace
+
+To perform generative queries, we have included a new `generate` namespace in the `v3` client. Through this namespace you have make queries as you would with the `query` namespace. 
```

Comment:
Consider:
The `v3` client adds a new namespace, `generate`. Use the `generate` like the `query` namespace to make queries.


QUESTION: Will TS user know what a generate namespace is? It would be helpful to add something like this:

The `v3` client adds a new namespace, `generate`. The `generate` namespace does ThisReallyCoolThing. 

Use the `generate` like the `query` namespace to make queries.